docs/concepts/iterable.md

```diff
@@ -3,170 +3,211 @@ title: Extracting Structured Data with Iterable and Streaming in Python
-
-```python
-import instructor
-import openai
-from typing import Iterable
-from pydantic import BaseModel
+    client = instructor.from_openai(openai.OpenAI())
 
-client = instructor.from_openai(openai.AsyncOpenAI(), mode=instructor.Mode.TOOLS)
+    class User(BaseModel):
+        name: str
+        age: int
 
+    resp = client.chat.completions.create_iterable(
+        model="gpt-4o-mini",
+        messages=[
+            {
+                "role": "user",
+                "content": "Ivan is 28, lives in Moscow and his friends are Alex, John and Mary who are 25, 30 and 27 respectively",
+            }
+        ],
+        response_model=User,
+    )
+
+    for user in resp:
+        print(user)
+    ```
+    _Recommended for most use cases. Handles streaming and iteration for you._
+
+=== "Using `create` with `Iterable[User]`"
+    ```python
+    import instructor
+    import openai
+    from pydantic import BaseModel
+    from typing import Iterable
+
+    client = instructor.from_openai(openai.OpenAI())
+
+    class User(BaseModel):
+        name: str
+        age: int
+
+    resp = client.chat.completions.create(
+        model="gpt-4o-mini",
+        messages=[
+            {
+                "role": "user",
+                "content": "Ivan is 28, lives in Moscow and his friends are Alex, John and Mary who are 25, 30 and 27 respectively",
+            }
+        ],
+        response_model=Iterable[User],
+    )
+
+    for user in resp:
+        print(user)
+    ```
+    _Use this if you need more manual control or compatibility with legacy code._
+
+---
+
+
+We also support more complex extraction patterns such as Unions as you'll see below out of the box. 
+
+??? warning
+
+    Unions don't work with Gemini because the AnyOf is not supported in the current response schema.
 
-class UserExtract(BaseModel):
-    name: str
-    age: int
+## Synchronous Usage
 
+=== "Using `create`"
 
-async def print_iterable_results():
-    model = await client.chat.completions.create(
+    ```python
+    import instructor
+    import openai
+    from typing import Iterable, Union, Literal
+    from pydantic import BaseModel
+
+    class Weather(BaseModel):
+        location: str
+        units: Literal["imperial", "metric"]
+
+    class GoogleSearch(BaseModel):
+        query: str
+
+    client = instructor.from_openai(openai.OpenAI(), mode=instructor.Mode.TOOLS)
+
+    results = client.chat.completions.create(
         model="gpt-4",
-        response_model=Iterable[UserExtract],
-        max_retries=2,
+        messages=[
+            {"role": "system", "content": "You must always use tools"},
+            {"role": "user", "content": "What is the weather in toronto and dallas and who won the super bowl?"},
+        ],
+        response_model=Iterable[Union[Weather, GoogleSearch]],
         stream=True,
+    )
+
+    for item in results:
+        print(item)
+    ```
+
+=== "Using `create_iterable` (recommended)"
```

Comment:
Remind me what the === for d

Reply:
It's for a content tab! This allows you to display two different variants of the code but you lose out on the ability to do code highlighting for specific lines 

https://github.com/user-attachments/assets/df7c0411-a88e-4e47-be67-b0f219446258